Backport changes to v4.3.x branch

xavier2k6 opened the pull request:
https://github.com/qbittorrent/qBittorrent/pull/14936
https://github.com/qbittorrent/qBittorrent/pull/15247
https://github.com/qbittorrent/qBittorrent/pull/15262
https://github.com/qbittorrent/qBittorrent/pull/15264
https://github.com/qbittorrent/qBittorrent/pull/15281
https://github.com/qbittorrent/qBittorrent/pull/15282
https://github.com/qbittorrent/qBittorrent/pull/15306 (3 of 4 commits) ref: https://github.com/qbittorrent/qBittorrent/pull/15304#issuecomment-902419491
https://github.com/qbittorrent/qBittorrent/pull/15339

glassez commented:
@xavier2k6 
Can you add #15282 here?

xavier2k6 commented:
> @xavier2k6
> Can you add #15282 here?

Done & OP Updated.

sledgehammer999 commented:
Can you also add #15262 ?
@Chocobo1  does it make sense to backport #15281 too?

xavier2k6 commented:
> Can you also add #15262 ?

Done & OP Updated.

xavier2k6 commented:
https://github.com/qbittorrent/qBittorrent/pull/15247?

Chocobo1 commented:
>@Chocobo1 does it make sense to backport #15281 too?
>
>#15247?

Should be fine to backport both.

ArcticGems commented:
Maybe also PR https://github.com/qbittorrent/qBittorrent/pull/15251??

sledgehammer999 commented:
>Maybe also PR #15251??

It's backported already and in v4.3.7. See 2b05b2b4711a128226e38c2637d5152180c57026

xavier2k6 commented:
Added https://github.com/qbittorrent/qBittorrent/pull/15281 & https://github.com/qbittorrent/qBittorrent/pull/15247 - OP Updated.

glassez commented:
@xavier2k6 
You shouldn't add any merge commits here.

xavier2k6 commented:
> You shouldn't add any merge commits here.

So, only the 5 commits that make up PR https://github.com/qbittorrent/qBittorrent/pull/15281??

glassez commented:
> > You shouldn't add any merge commits here.
> 
> So, only the 5 commits that make up PR #15281??

@Chocobo1, @sledgehammer999 
What do you think? Is such inner merge commit acceptable? IMO, there is no problem if we apply "rebase and merge" to this PR.

sledgehammer999 commented:
Up until now the tradition is to `cherry-pick` individual commits into the stable branch. I'd like to keep it that way.

glassez commented:
> Up until now the tradition is to `cherry-pick` individual commits into the stable branch. I'd like to keep it that way.

Moreover, we are now trying to avoid creating "merge" commits even for regular PRs.

PriitUring reviewed (commented):
Shouldn't title have "file health" instead of "file heath"?

xavier2k6 commented:
> Shouldn't title have "file health" instead of "file heath"?

That's the way it was originally merged to master.

Chocobo1 commented:
> > Shouldn't title have "file health" instead of "file heath"?
> 
> That's the way it was originally merged to master.
> 
> @Chocobo1 ?

You may revise the commit message if you want.

sledgehammer999 commented:
>>That's the way it was originally merged to master.
>You may revise the commit message if you want.

Better not. Someone looking at the commit in the `v4_3_x` should be able to find it in `master` by searching by commit title.

xavier2k6 commented:
>Better not. Someone looking at the commit in the `v4_3_x` should be able to find it in `master` by searching by commit title.

Ok, I will leave it as-is.

glassez commented:
@sledgehammer999, @Chocobo1 
I would also suggest to put PR number into commit body (instead of commit subject) when we use `Squash and merge` way. Then history will look more consistent.

xavier2k6 commented:
Add PR https://github.com/qbittorrent/qBittorrent/pull/15264?

sledgehammer999 commented:
To anyone involved please finalize this PR. I want to do a release this weekend.

xavier2k6 commented:
> Add PR #15264?

Added

>To anyone involved please finalize this PR. I want to do a release this weekend.

I'm done with this now.

Chocobo1 commented:
@xavier2k6 
This commit ("Add tooltip to listening port spinbox") shouldn't backport as it is part of other changes that isn't suitable for v4_3_x, the other commits are OK.
At the same time, you can backport #15339 if you want.

xavier2k6 commented:
> This commit ("Add tooltip to listening port spinbox") shouldn't backport as it is part of other changes that isn't suitable for v4_3_x

Removed

> At the same time, you can backport #15339 if you want.

Added & OP updated.

Chocobo1 commented:
@xavier2k6 
Thanks!
